receiver/statsdreceiver/transport/udp_server.go

```diff
@@ -6,47 +6,50 @@ package transport // import "github.com/open-telemetry/opentelemetry-collector-c
 import (
 	"bytes"
 	"errors"
+	"fmt"
 	"io"
 	"net"
 	"strings"
 
 	"go.opentelemetry.io/collector/consumer"
-
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/protocol"
 )
 
 type udpServer struct {
 	packetConn net.PacketConn
-	reporter   Reporter
+	transport  Transport
 }
 
+// Ensure that Server is implemented on UDP Server.
 var _ (Server) = (*udpServer)(nil)
 
 // NewUDPServer creates a transport.Server using UDP as its transport.
-func NewUDPServer(addr string) (Server, error) {
-	packetConn, err := net.ListenPacket("udp", addr)
-	if err != nil {
-		return nil, err
+func NewUDPServer(transport Transport, address string) (Server, error) {
+	var usrv udpServer
+	var err error
+
+	if !transport.IsPacketTransport() {
+		return nil, ErrUnsupportedPacketTransport
 	}
 
-	u := udpServer{
-		packetConn: packetConn,
+	usrv.transport = transport
+	usrv.packetConn, err = net.ListenPacket(transport.String(), address)
+	if err != nil {
+		return nil, fmt.Errorf("starting to listen %s socket: %w", transport.String(), err)
 	}
-	return &u, nil
+
+	return &usrv, nil
```

Comment:
super nit, I prefer returning directly:
```go
return &udpServer{
   usrv.transport = transport
   usrv.packetConn = packetConn
}, nil
```

In this way you do not have to define the error and udpServer at the begininng and I think it is more readable

Reply:
Done.